### PR TITLE
Modifier refactor

### PR DESCRIPTION
**Pull request by Protonull**

### CODE REVIEW

**Comment by Protonull** [comment]
This relies on the APIs added int [this PR](https://github.com/CivClassic/CivModCore/pull/40)

**Comment by Protonull** [comment]
This PR probably seems like a lot, but most of it cascade. Remember that Map I showed you that mapped 1:1 a class and an instance of that class? Well I refactored that to be less generic based, but it also meant making the modifiers themselves less generic based, so most of the changes to the modifier classes are just changing the signature of the classes.

Additionally, as described in the two commits labelled "config change", there's been two substantial removals from the config, thus also a removal from the config parser and the modifiers that used those config methods... again.. lots of cascade :P 

What's the actual takeaway from all these changes?

  1. ItemExchange now works natively with my CustomItems system that's currently in PR

  2. The successful transaction button feature is functional

However there are some consequences to these changes:

  1. The config in the AnsibleSetup will need to be changed to reflect [these changes](https://github.com/CivClassic/ItemExchange/pull/14/commits/acb57536b068d61e5ae2a6ebcaa37046a2ce2c02#diff-ee97532d2d40af21fc87a8fc24cd37c3R12-R19)

  2. (POTENTIAL) I did what you said and refactored the `IETransactionEvent`. You said that it could be removed since the only plugin that used it was JukeAlert and it doesn't anymore, but I still want to emit such an event, so I've instead refactored it. I'm putting this warning here just in case some other plugin also listens for `IETransactionEvent`.

Basically the PR looks scary but the amount of actual changes in what's being done is quite small. Also, the reason why these commits keep failing is because like with takeaway point one, there are a few calls to new APIs that are currently in PR.

File: src/main/java/com/untamedears/itemexchange/rules/modifiers/_ExampleModifier.java [hunk 1]
```diff
@@ -15,17 +15,10 @@
 
 @CommandAlias(SetCommand.ALIAS) // This is needed to make commands work
 @Modifier(slug = "EXAMPLE", order = 12345)
-public final class _ExampleModifier extends ModifierData<_ExampleModifier> {
+public final class _ExampleModifier extends ModifierData {
```
Comment (Maxopoly) [inline src/main/java/com/untamedears/itemexchange/rules/modifiers/_ExampleModifier.java:18]: The heck is with that class name

Comment (Protonull) [reply src/main/java/com/untamedears/itemexchange/rules/modifiers/_ExampleModifier.java:18]: This class isn't used at all, it's just an example, and the `_` is to ensure it's at the top of the folder. When creating a new Modifier, I just copy paste this class and rename it.
